src/react-imgix.js

```diff
@@ -63,43 +39,89 @@ export default class ReactImgix extends Component {
     disableLibraryParam: PropTypes.bool
   };
   static defaultProps = {
-    aggressiveLoad: false,
     auto: ["format"],
     entropy: false,
     faces: true,
     fit: "crop",
-    fluid: true,
-    generateSrcSet: true,
+    disableSrcSet: false,
     onMounted: () => {},
-    precision: 100,
     type: "img"
   };
-  state = {
-    width: null,
-    height: null,
-    mounted: false
-  };
 
-  forceLayout = () => {
+  componentDidMount = () => {
     const node = ReactDOM.findDOMNode(this);
-    this.setState({
-      width: node.scrollWidth,
-      height: node.scrollHeight,
-      mounted: true
-    });
     this.props.onMounted(node);
   };
 
-  componentDidMount = () => {
-    this.forceLayout();
-  };
+  buildSrcs = () => {
+    const props = this.props;
+    const {
+      width,
+      height,
+      entropy,
+      faces,
+      auto,
+      customParams,
+      disableLibraryParam,
+      disableSrcSet,
+      type
+    } = props;
+
+    let crop = false;
+    if (faces) crop = "faces";
+    if (entropy) crop = "entropy";
+    if (props.crop) crop = props.crop;
+
+    let fit = false;
+    if (entropy) fit = "crop";
+    if (props.fit) fit = props.fit;
 
-  _findSizeForDimension = dim =>
-    findSizeForDimension(dim, this.props, this.state);
+    const fixedSize = width != null || height != null;
+
+    const srcOptions = {
+      auto,
+      ...customParams,
+      crop,
+      fit,
+      ...(disableLibraryParam ? {} : { ixlib: `react-${PACKAGE_VERSION}` }),
+      ...(fixedSize && height ? { height } : {}),
+      ...(fixedSize && width ? { width } : {})
+    };
+
+    const src = constructUrl(this.props.src, srcOptions);
+
+    let srcSet;
+
+    if (disableSrcSet) {
+      srcSet = src;
```

Comment:
Hmm... the `disabledSrcSet` option doesn't actually disabled `srcset`. Is that bad? Maybe there's no good reason not to just set it anyway as we're doing here, but if the option says "don't set it" the user expectation might be that it doesn't get set. I don't feel strongly either way, but it's food for thought.